src/languageclient.ts

```diff
@@ -0,0 +1,281 @@
+                        if (!capabilities) {
+                            return vscode.window.showErrorMessage('The language server is not able to serve any features at the moment.');
+                        } else {
+
+                            progress.report({
+                                message: "Language Server Ready",
+                                increment: 100
+                            });
+                            ExperimentalLanguageClient.isRunning = true;
+
+                            resolve();
+                        }
+                    });
+
+                    progress.report({
+                        message: "Initializing Language Server",
+                        increment: 70
+                    });
+
+                    // Start the client. This will also launch the server
+                    this.ctx.subscriptions.push(langClient.start());
+
+                    try {
+                        await this.checkForUpdatedLsp();
+                    } catch (e) {
+                        return vscode.window.showErrorMessage(`Failed checking for newer language server version. Error: ${e}`);
+                    }
+
+                } catch (e) {
+                    vscode.window.showErrorMessage(`Failed starting language server. Error: ${e}`);
+                    reject(e);
+                }
+            });
+        });
+
+    }
+
+    private async checkForUpdatedLsp() {
+        if (this.shouldCheckForLastestRelease()) {
+            // After server is running lets do a check to see if a newer one exists
+            const octokit = new Octokit();
+            const release = await octokit.repos.getLatestRelease({
+                owner: 'juliosueiras',
+                repo: 'terraform-lsp'
+            });
+            if (this.currentReleaseId !== release.data.id.toString()) {
+                vscode.window.showInformationMessage(`A newer Lanaguage Server is available ${release.data.tag_name}. Run the 'Install/Update Language Server' command to update`);
+            }
+        }
+    }
+
+    private async installCommonProviders(serverLocation: string) {
+        let terminal = vscode.window.createTerminal("Terraform: Install providers");
+        const defaultProvidersPath = Path.join(serverLocation, "providers.tf");
+        if (!fs.existsSync(defaultProvidersPath)) {
+            fs.writeFileSync(defaultProvidersPath, `
+provider "aws" {}
+provider "azurerm" {}
+provider "google" {}
+provider "alicloud" {}
+provider "helm" {}
+provider "kubernetes" {}
+provider "random" {}
+provider "null" {}
+provider "external" {}
+provider "template" {}
+provider "archive" {}
+                        `);
+        }
+        const defaultProvidersPluginsPath = Path.join(serverLocation, ".terraform", "plugins");
+        if (!fs.existsSync(defaultProvidersPluginsPath)) {
+            terminal.show();
+            terminal.sendText(`cd ${serverLocation}`);
```

Comment:
The server location url needs to be quoted here; otherwise it can fail for invalid paths (e.g. it fails on Windows if the user name contains a space).